application_form/tests/test_lottery_api.py

```diff
@@ -36,6 +36,30 @@ def test_execute_lottery_for_project_post_badly_formatted_project_uuid(api_clien
     assert response.status_code == status.HTTP_400_BAD_REQUEST
 
 
+@pytest.mark.django_db
+@pytest.mark.usefixtures("elastic_apartments")
+def test_execute_lottery_for_project_post_fails_application_time_not_finished(
+    api_client, elastic_apartment_application_time_active
+):
+    es_project = elastic_apartment_application_time_active
+    profile = SalespersonProfileFactory()
+    api_client.credentials(HTTP_AUTHORIZATION=f"Bearer {_create_token(profile)}")
+
+    project = ProjectFactory(uuid=es_project.project_uuid)
+    apartment = ApartmentFactory(
+        project=project, project__ownership_type=OwnershipType.HITAS
+    )
+    app = ApplicationFactory(type=ApplicationType.HITAS)
+    app.application_apartments.create(apartment=apartment, priority_number=0)
+    add_application_to_queues(app)
+
+    data = {"project_uuid": apartment.project.uuid}
+    response = api_client.post(
+        reverse("application_form:execute_lottery_for_project"), data, format="json"
+    )
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
```

Comment:
There can be a ton of reasons why a request gets HTTP 400 from DRF, so to make sure this test isn't passing falsely, I think it could be nice if the response was somehow verified to be the right one. Maybe check the error message also?

Reply:
Added message check.